internal/emlibc/resolver.go

```diff
@@ -0,0 +1,114 @@
+package emlibc
```

Comment:
Should this be internal? I imagine some embedded users would want to call `GetEnv()`.

Reply:
So I was trying to decide how to handle that so i put it in there as more of an afterthought.

I may be overthinking it but...

should we be laying the ground work for supporting multiple 'environments' (EM, WASI, the other 15 competing specifications that are sure to come)

Should they be built in like i've started, or should they themselves be external wasm files that use a common libc like api that we develop internally. 

So at the moment I'm implementing it as a ResolveFunc. 

I was thinking about making the ReadModule signature variadic
```go
ReadModule(r io.Reader, resolvePath ...ResolveFunc) (*Module, error) 
```
but i'm not sure if thats the correct way to go about it so that someone could call

```go
wasm.ReadModule(buf,EMLibc,WASI,FileImporter,WAPM)
```
and we go through each in turn to try and resolve 

 some of that can be figured out later...but I was questioning if I was hooking in at the right place.

Reply:
Hmmmmm, good point ... We should have a think about this.

I think the 'resolver' method is probably the right way to go. I'm not a fan of inventing our own internal API, as thats another abstraction layer to maintain and might affect performance depending on implementation.